.github/workflows/broadcast-frontend-hash.yml

```diff
@@ -0,0 +1,78 @@
+name: Broadcast Frontend Hash
+on:
+  workflow_dispatch:
+    inputs:
+      dfx_version:
+        description: 'Release version of dfx'
+        default: "latest"
+  # release:
+  #   # https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads?actionType=released#release
+  #   # "A release was published, or a pre-release was changed to a release."
+  #   types: [released]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  PLAYGROUND_REPO: ${{ github.repository_owner }}/motoko-playground
+  FILE: service/wasm-utils/whitelisted_wasms.txt
+
+jobs:
+  update-frontend-hash:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout dfinity/sdk repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # workaround to fetch all tags: https://github.com/actions/checkout/issues/701
+          path: sdk
+
+      - name: Get new hash
+        working-directory: ./sdk
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          git checkout ${{ inputs.dfx_version }}
+          echo "NEW_HASH=$(shasum -a 256 src/distributed/assetstorage.wasm.gz | cut -f1 -d" ")" >> $GITHUB_ENV
+
+      - name: Checkout dfinity/motoko-playground repo
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.NIV_UPDATER_TOKEN }}
+          repository: ${{ env.PLAYGROUND_REPO }}
+          path: motoko-playground
+
+      - name: Modify Rust file
+        working-directory: ./motoko-playground
+        run: |
+          tempfile=$(mktemp)
+          # `$i` specifies that the insertion should happen at the line before the last line. `$` matches the last line, and `i` inserts before the matched line.
+          # `\/\/` is `//` escaped for sed, same for `\"` -> `"`, and `\ ` -> ` `
+          sed '$i\ \ \ \ \"${{ env.NEW_HASH }}\", \/\/ dfx ${{ inputs.dfx_version }} frontend canister' "${{ env.FILE }}" > "$tempfile"
+          # not worth it to install sponge
+          mv "$tempfile" "${{ env.FILE }}"
```

Comment:
For reference: https://github.com/dfinity/motoko-playground/pull/185

First, let's look at https://github.com/dfinity/motoko-playground/blob/main/service/wasm-utils/whitelisted_wasms.txt
```
[
    "88d1e5795d29debc1ff56fa0696dcb3adfa67f82fe2739d1aa644263838174b9", // dfx 0.15.0 frontend canister
]
```

The extension of `whitelisted_wasms.txt` is `.txt`, which communicates that it's a text file.  But it contains what looks like JSON array brackets.  However, it also contains a comment, so maybe its content is json5.

But continuing, `fn is_whitelisted(wasm)` just looks for the hash anywhere in the file:
```
    let white_list = include!("whitelisted_wasms.txt");
    if white_list.contains(&wasm_hash.as_str()) {
```

My recommendation is to revise that whitelist file to be an actual text file, containing no JSON brackets.  Its format is, per line: a hash, followed optionally by a description.

```
88d1e5795d29debc1ff56fa0696dcb3adfa67f82fe2739d1aa644263838174b9 dfx 0.15.0 frontend canister
```

That works fine with the existing method of looking for the hash anywhere in the file, and adding a new entry will be trivial:

```suggestion
          echo "${{ env.NEW_HASH }} dfx ${{ inputs.dfx_version }} frontend canister" >> "${{ env.FILE }}"
```

Reply:
The whitelisted_wasms.txt was accepted by the repo owners. I'd like to avoid going round and round; could you @chenyan-dfinity please take a look and tell us what you think?

Reply:
That txt file is a Rust code snippet, not json. What's the motivation for this change? We don't have to use `sed` here, `head -n -1` would do the work.

Reply:
Got it.  I missed the distinction between `include!("whitelisted_wasms.txt")` and `include_str!("whitelisted_wasms.txt")`.  The motivation is to have some reasonable way to add to the contents.  If we can get there with `head` rather than `sed`, then all good.

Reply:
The behavior of the `head` command differs between the GNU and BSD versions. The command `head -n -1` will work on the GNU version of `head` but will not work as expected on the BSD version (which is the one included on macOS). GH workflow is based on `ubuntu-latest` so it will work there, however, the proposed `sed` solution works on both MacOS and Linux, making it debuggable and easier to work with in my opinion. Leaving the format of the `.txt` file aside, I don't understand what's so bad about `sed` solution, in my opinion, the only unclear thing there was `$i` which behavior of I documented in the comment above it. 

Here is `head`-based solution that works on both OSes:
```bash
head -n $(($(wc -l < whitelisted_wasms.txt) - 1)) whitelisted_wasms.txt > tmp
echo "    \"${{ env.NEW_HASH }}\", \/\/ dfx ${{ inputs.dfx_version }} frontend canister\n]" >> tmp
mv -f tmp whitelisted_wasms.txt
```

Reply:
Recommendation: go with `head -n -1`:
- It's simpler than either the `sed` or `head(math(wc))` solutions
- It works on `ubuntu-latest` where the workflow runs
- A github workflow isn't so "debuggable" with `gh workflow run` if you need to be concerned about differences between the local environment and where it `runs-on`, because what does `gh workflow run` prove in that case?

> don't understand what's so bad about sed solution

A variant on an old joke: "A software engineer, confronted with a problem, thinks "I know, I'll use `sed`." Now he has two problems."

Also maintenance.  Sometime in the future, someone else will have to understand how this works.

With more experience will come a more productive response to `sed '$i\ \ \ \ \"${{ env.NEW_HASH }}\", \/\/ `: "I'll find a different way to do this."